ui/components/component-library/checkbox/README.mdx

```diff
@@ -53,9 +53,57 @@ Use the `isIndeterminate` boolean prop to set the indeterminate state of the `Ch
 </Canvas>
 
 ```jsx
-import { Checkbox } from '../../component-library';
+import React from 'react';
+import { Checkbox, Box } from '../../component-library';
+
+const [checkedItems, setCheckedItems] = React.useState([false, true, false]);
+
+const allChecked = checkedItems.every(Boolean);
+const isIndeterminate = checkedItems.some(Boolean) && !allChecked;
+
+const handleIndeterminateChange = () => {
+  if (allChecked || isIndeterminate) {
+    setCheckedItems([false, false, false]);
+  } else {
+    setCheckedItems([true, true, true]);
+  }
+};
 
-<Checkbox isIndeterminate={true} />;
+const handleCheckboxChange = (index, value) => {
+  const newCheckedItems = [...checkedItems];
+  newCheckedItems[index] = value;
+  setCheckedItems(newCheckedItems);
+};
+
+<Checkbox
+    label="isIndeterminate demo"
+    isChecked={allChecked}
+    isIndeterminate={isIndeterminate}
+    onChange={handleIndeterminateChange}
+    marginBottom={2}
+/>
+<Box
+  marginLeft={2}
+  gap={1}
+  display={Display.Flex}
+  flexDirection={FlexDirection.Column}
+>
+  <Checkbox
+    isChecked={checkedItems[0]}
+    onChange={(e) => handleCheckboxChange(0, e.target.checked)}
+    label="Checkbox 1"
+  />
+  <Checkbox
+    isChecked={checkedItems[1]}
+    onChange={(e) => handleCheckboxChange(1, e.target.checked)}
+    label="Checkbox 2"
+  />
+  <Checkbox
+    isChecked={checkedItems[2]}
+    onChange={(e) => handleCheckboxChange(2, e.target.checked)}
+    label="Checkbox 3"
+  />
+</Box>
```

Comment:
Provides code example to copy paste for engineers

<img width="461" alt="Screenshot 2023-07-13 at 2 06 38 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/5140f846-ac5b-4670-befc-f954ec9bdd25">